### ISSUE TITLE
The framework "ignores" the setState function calls

### ISSUE THREAD

**Issue by limonadev** [peiffer-innovations/automated_testing_framework#5]
## Steps to reproduce
1. Clone [this](https://github.com/limonadev/tutorial/tree/main/automated_testing_medium) repository.
2. Run it.

## Expected behavior
The framework runs the tests and after the increment button is pressed, the `Text` widget shows the new `_counter` value. As the `Text` widget has changed its value, the framework reads this new updated value as expected.

## Actual behavior
The framework when trying to detect the new value of the `Text` receives instead the old `_counter` value.

**Comment by jpeiffer** [comment]
Found the issue.  This one was a bit of a fun one to figure out.

The `Testable` widget was creating "locked" references to the old widget when it was generating the common functions.  The fix was pretty straight forward.  Adding in this function to state object within `Testable` fixed it:

```
  @override
  void didUpdateWidget(Widget oldWidget) {
    super.didUpdateWidget(oldWidget);

    _types.remove(TestableType.error_requestable);
    _types.remove(TestableType.value_requestable);
    _types.remove(TestableType.value_settable);

    _onRequestError =
        widget.onRequestError ?? _tryCommonGetErrorMethods(widget.child);
    if (_onRequestError != null) {
      _types.add(TestableType.error_requestable);
    }

    _onRequestValue =
        widget.onRequestValue ?? _tryCommonGetValueMethods(widget.child);
    if (_onRequestValue != null) {
      _types.add(TestableType.value_requestable);
    }

    _onSetValue = widget.onSetValue ?? _tryCommonSetValueMethods(widget.child);
    if (_onSetValue != null) {
      _types.add(TestableType.value_settable);
    }
  }
```

I'll try to get this deployed tonight with my next framework release.

**Comment by jpeiffer** [comment]
The fix has been deployed as `1.0.3`, please retest when you can.

**Comment by limonadev** [comment]
Confirmed, the bug was fixed.
